### ISSUE TITLE
StateGoal should use canonical state as function key

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1017]
In GitLab by @jvande42b on Apr 9, 2018, 17:10

when setting the function key in state goal, it should do something like this:

```python
canonical, sign = optimization_problem.alias_relation.canonical_signed(state)
self.function_key = canonical if sign > 0.0 else '-' + canonical
```

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 11, 2018, 21:30

mentioned in commit fd06871ae0b247575ae75944628f38698586628e

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 11, 2018, 21:53

mentioned in commit 3830ba35cd949fbded20e58cf0390249a94f1f1e

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 16, 2018, 10:05

mentioned in commit 242cce0b6d903dd23df48fb3880d7788a2668843

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 16, 2018, 13:17

mentioned in commit 7573db8b84ab9f2c88d280cfa953bad18f4c988c

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 16, 2018, 13:29

mentioned in commit 02287638d88c96bd292e9d2f081b0e41c00b6c13

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 16, 2018, 14:00

mentioned in commit 44522cf16d1e71f283d10067ae91b05e81ac2c17

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 16, 2018, 14:52

closed via commit 44522cf16d1e71f283d10067ae91b05e81ac2c17

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 16, 2018, 14:52

closed via merge request !111
